### DISCUSSION TITLE
Memory  increased

### DISCUSSION THREAD

**Discussion by EricVision** [LiorBanai/HDF5-CSharp#222] [Q&A] [answered]
HI  LiorBanai：
      Thank you for Reading!
      I Have a problem need your help.
     The problem is when i use  the function of "Hdf5.ReadObject", my computer memory will be occupied more and it will not down.  Could you give me some advice to solve this problem?

                        for (int i = 0; i < groupCount; i++)
                        {
                            HDF5DataClass dataClass = new HDF5DataClass();
                            dataClass = Hdf5.ReadObject<HDF5DataClass>(fileID, $"/{sortGroup[i]}");
                            temp[i] = dataClass.Location;   
                        }  
                      internal class HDF5DataClass
                       {
                                public double Location { get; set; }
                               public byte[] Image { get; set; }

                        }

**Comment by LiorBanai** [comment]
Hi @EricVision ,
I need more information to investigate this .
- What version of HDF-CSharp are you using?
- what is the file size and the memory footprint?
- what is the average size of the byte array?
- how many groups are you reading in the loop?

I'll investigate after that :)

**Reply by EricVision** [reply]
Hi:

- The Version is 1.14.3
- The  size if the .h5 file is about 860M, and the RAM will increase 30M
- The size of the byte array is about 30000 
- There have 3000 groups.
-

**Reply by LiorBanai** [reply]
@EricVision Thanks, I'll try to reproduce it on my side and update you

**Reply by EricVision** [reply]
Thank you very much!

**Reply by LiorBanai** [reply]
@EricVision  I believe I have found the issue. I'll release next week new version.

Thanks for reporting this

**Comment by LiorBanai** [comment, marked as answer]
Release new version
https://www.nuget.org/packages/HDF5-CSharp/1.15.3

please check this one and report if there are still issues
